arango/utils.py

```diff
@@ -82,3 +82,27 @@ def is_none_or_str(obj: Any) -> bool:
     :rtype: bool
     """
     return obj is None or isinstance(obj, str)
+
+
+def get_batches(
+    l: Sequence[Json], batch_size: Optional[int] = None
+) -> Union[List[Sequence[Json]], Iterator[Sequence[Json]]]:
+    """Generator to split a list in batches
+        of (maximum) **batch_size** elements each.
+        If **batch_size** is invalid, return entire
+        list as one batch.
+
+    :param l: The list of elements.
+    :type l: list
+    :param batch_size: Number of elements per batch.
+    :type batch_size: int | None
+    """
+    if batch_size is None or batch_size <= 0 or batch_size >= len(l):
+        return [l]
+
+    def generator() -> Iterator[Sequence[Json]]:
+        n = int(batch_size)  # type: ignore # (false positive)
+        for i in range(0, len(l), n):
+            yield l[i : i + n]
+
+    return generator()
```

Comment:
I think it's reasonable to assume that when users are calling this function, they know the `batch_size` in advance (so we don't have to make it optional). And the code can be simplified to something like this:
```
def get_batches(elements: Sequence[Json], batch_size: int) -> Iterator[Sequence[Json]]]:
    for index in range(0, len(elements), batch_size):
        yield elements[index: index + batch_size]
```

Reply:
makes sense, assuming we move to `import_bulk_by_batches`

Reply:
I agree with @joowani . Does it make sense to flag an error if this is not provided?

Reply:
> makes sense, but I assume this change is required only if we move to `import_bulk_by_batches`? Otherwise we would need to retain the current state of `get_batches` to support `batch_size=None`

Not sure if I understood, but we would be leaving `import_bulk` as is, and `batch_size` would be a required argument in `import_bulk_by_batches`.